andela/Ah-backend-guardians
# #162685665 User can rate articles

#### What does this PR do?
Users should be able to rate articles that have been created. 

#### Description of Task to be completed?

- Users should be able to rate articles
- Users cannot rate articles they have created
- The average rating of an article should be returned based on the total number of ratings an article has


#### What are the relevant pivotal tracker stories?
[#162685665](https://www.pivotaltracker.com/n/projects/2232496)

#### How should this be tested remotely?
The following endpoints can be hit to show how the app works remotely

```
GET : https://dashboard.heroku.com/apps/ah-backend-guardians-sta-pr-29//api/articles/<slug>/

PUT : https://dashboard.heroku.com/apps/ah-backend-guardians-sta-pr-29//api/articles/<slug>/ratings

```

#### How should this be locally tested and reviewed?
Before testing run these commands in cmd

`git clone https://github.com/andela/Ah-backend-guardians.git`

`cd Ah-backend-guardians`

`git checkout ft-rate-an-article-162685665`

`virtualenv -python=python3 env`

`. env/bin/activate`

`pip install -r requirements.txt`

1.  Create a postgres database and a `.env` file in your root directory and create a `.env` based on the `.env.example` file with the necessary credentials. 

2. Open Postman and type `http://localhost:8000/api/articles/<slug>/` in url text box. Select GET in methods drop down then click `SEND` button to display a singular article and its rating based on the slug. A sample of the expected output is shown below
```
{
    "article": {
        "id": 3,
        "title": "Andela is great",
        "body": "Andela rocks",
        "description": "naahh, its great",
        "author": {
            "email": "nserekopaul@gmail.com",
            "username": "myrdstom"
        },
        "slug": "andela-is-great-2",
        "published": true,
        "created_at": "2019-01-24T18:32:24.446758Z",
        "updated_at": "2019-01-24T18:32:24.446778Z",
        "images": "",
        "read_time": 0,
        "tags": [],
        "average_rating": 1
    }
}
```



#### Accompanying screenshot of how it should look like

<img width="1122" alt="screen shot 2019-01-22 at 7 15 10 pm" src="https://user-images.githubusercontent.com/30922279/51549050-a71c3d80-1e7a-11e9-8413-e32d2cb4a8f3.png">

3. Type `http://localhost:8000/api/articles/<slug>/rating/` in url text box. Select POST in methods  
 drop-down and populate the data with a template of the information below:
```
{
   "score":5
}
```
then click `SEND` button to post a rating

#### Screenshot:
<img width="1122" alt="screen shot 2019-01-22 at 7 18 43 pm" src="https://user-images.githubusercontent.com/30922279/51549040-a4214d00-1e7a-11e9-819a-55e37910ce3d.png">
